js/interface.js

```diff
@@ -1305,30 +1325,33 @@ function uploadFiles(files) {
 
       hideProgressBar();
 
-      return getApps({
-        cache: false
-      });
-    })
-    .then(function (apps) {
-      // Updated list to update the storage usage values
-      userApps = apps;
-
-      var value = $fileDropDown.val().split('_');
-      var appId = parseInt(value[1], 10);
+      if (!appId) {
+        return;
+      }
 
+      return updateAppMetrics(appId);
+    })
+    .then(function() {
       toggleStorageUsage(appId);
     })
     .catch(handleUploadingError);
 }
 
-function handleUploadingError() {
+function handleUploadingError(error) {
   hideProgressBar();
+
   if (isCancelClicked) return;
-  showError();
+
+  showError(Fliplet.parseError(error));
```

Comment:
I would provide the fallback to the parser (because it's scenario specific). And that way the `showError()` function doesn't need a default message (because it'll then purely be purposed to show the message, whatever that's given).

```suggestion
  showError(Fliplet.parseError(error, 'There was an error while uploading your file. Please try again.'));
```